hrw.go

```diff
@@ -1,280 +1,118 @@
 // Package hrw implements Rendezvous hashing.
-// http://en.wikipedia.org/wiki/Rendezvous_hashing.
+// https://en.wikipedia.org/wiki/Rendezvous_hashing.
 package hrw
 
 import (
-	"encoding/binary"
-	"errors"
-	"math"
-	"reflect"
 	"sort"
 
 	"github.com/twmb/murmur3"
+	"golang.org/x/exp/constraints"
 )
 
-type (
-	// Hasher interface used by SortSliceByValue.
-	Hasher interface{ Hash() uint64 }
+// Hashable is something that can be hashed.
+type Hashable interface{ Hash() uint64 }
 
-	sorter struct {
-		l    int
-		less func(i, j int) bool
-		swap func(i, j int)
-	}
-)
-
-// Boundaries of valid normalized weights.
-const (
-	NormalizedMaxWeight = 1.0
-	NormalizedMinWeight = 0.0
-)
+// HashableSlice implements Hashable interface over
+// raw data. Use [WrapBytes] to instantiate correct
+// byte slice wrapper.
+type HashableSlice []byte
```

Comment:
I'd call it `HashableBytes` because a `Slice` can be treated as a slice of anything, but it's not critical.